versioned/spi/src/main/java/org/projectnessie/versioned/StoreWorker.java

```diff
@@ -43,6 +50,16 @@
 
   boolean requiresGlobalState(CONTENT content);
 
+  default boolean requiresPerContentState(ByteString content) {
+    return requiresPerContentState(getType(content));
+  }
+
+  default boolean requiresPerContentState(CONTENT content) {
+    return requiresPerContentState(getType(content));
+  }
+
+  boolean requiresPerContentState(Enum<CONTENT_TYPE> contentType);
```

Comment:
Why not just `CONTENT_TYPE`?

Reply:
Generics bites with `boolean requiresAttachments(CONTENT content)`